lib/options.nix

```diff
@@ -76,173 +76,238 @@ rec {
     );
   mkNullOrStrLuaFnOr = type: description: mkNullOrStrLuaFnOr' { inherit type description; };
 
-  defaultNullOpts = rec {
-    /**
-      Build a description with a plugin default.
-
-      The [default] can be any value, and it will be formatted using `lib.generators.toPretty`.
-
-      If [default] is a String, it will not be formatted.
-      This behavior will likely change in the future.
-
-      # Example
-      ```nix
-      mkDesc 1 "foo"
-      => ''
-        foo
-
-        Plugin default: `1`
-      ''
-      ```
-
-      # Type
-      ```
-      mkDesc :: Any -> String -> String
-      ```
-
-      # Arguments
-      - [default] The plugin's default
-      - [desc] The option's description
-    */
-    mkDesc =
-      default: desc:
-      let
-        # Assume a string default is already formatted as intended,
-        # historically strings were the only type accepted here.
-        # TODO deprecate this behavior so we can properly quote strings
-        defaultString = if isString default then default else generators.toPretty { } default;
-        defaultDesc =
-          "Plugin default:"
-          + (
-            # Detect whether `default` is multiline or inline:
-            if hasInfix "\n" defaultString then "\n\n```nix\n${defaultString}\n```" else " `${defaultString}`"
-          );
-      in
-      if desc == "" then
-        defaultDesc
-      else
+  defaultNullOpts =
+    let
+      # Convert `defaultNullOpts`-style arguments into normal `mkOption`-style arguments,
+      # i.e. moves `default` into `description` using `defaultNullOpts.mkDesc`
+      convertArgs =
+        { default, description, ... }@args:
+        (
+          args
+          // {
+            default = null;
+            description = defaultNullOpts.mkDesc default description;
+          }
+        );
+    in
```

Comment:
Putting this in a `let` block has caused the rest of the file to be indented. Maybe review with whitespace changes hidden...

Or we could come up with a better name for the helper and expose it publicly (removing the let-block).

Reply:
That said, almost all lines were touched _anyway_, so the extra indent probably isn't a big deal :thinking: